test/test_mkldnn.py

```diff
@@ -75,18 +76,38 @@ def test_detach(self):
         root = torch.randn(4, 5, dtype=torch.float32).to_mkldnn().requires_grad_()
 
         detach = root.detach()
-        self.assertEquals((4, 5), detach.size())
+        self.assertEqual((4, 5), detach.size())
         self.assertFalse(detach.requires_grad)
         self.assertTrue(root.requires_grad)
 
         detach_ = root.detach_()
-        self.assertEquals((4, 5), detach_.size())
+        self.assertEqual((4, 5), detach_.size())
         self.assertFalse(detach_.requires_grad)
         self.assertFalse(root.requires_grad)
 
     def test_repr(self):
         self.assertTrue("layout=torch._mkldnn" in str(torch.randn((1, 2, 3, 4),
                         dtype=torch.float, device=torch.device('cpu')).to_mkldnn()))
 
+    def test_conv2d(self):
+        N = torch.randint(3, 10, (1,)).item()
+        C = torch.randint(3, 10, (1,)).item()
+        M = torch.randint(3, 10, (1,)).item()
+        x = torch.randn(N, C, 224, 224, dtype=torch.float32) * 100
+
+        for bias in [True, False]:
+            conv2d = torch.nn.Conv2d(in_channels=C,
+                                     out_channels=M,
+                                     kernel_size=3,
+                                     stride=2,
+                                     padding=1,
+                                     bias=bias).float()
+            mkldnn_conv2d = copy.deepcopy(conv2d).to_mkldnn()
+
+            with torch.no_grad():
```

Comment:
why do you need the no_grad?

Reply:
Lol didn't you ask me to use to_grad in an comment above? https://github.com/pytorch/pytorch/pull/19075#discussion_r273675706

Reply:
Sorry for not being clear.

The recommendation was to use `no_grad` instead of `.data` -- that's just a general recommendation.

As for `.data` / `no_grad` itself, I'm trying to understand if you actually need it.  Do you?

Reply:
No it is not needed. Plus I now have added a to_mkldnn function at Module and used `_apply` to call to_mkldnn which handles the grad/.data thing well. I will remove the no_grad here.